### DISCUSSION TITLE
Sharing our feedback on the latest proposal

### DISCUSSION THREAD

**Discussion by CodeWalletTeam** [kinfoundation/kin-rfcs#169] [KRE 4.0 Discussion]
We at Code are pleased to submit these comments to the Kin Foundation’s Proposal for KRE v.4.0. Code is a new crypto wallet app that makes Kin simple to use. At the outset, we appreciate that the Kin Foundation submitted this Proposal. We view the Kin Foundation as an important partner in the Ecosystem, with a unique perspective on the functioning of the Kin Rewards Engine. 

Our comments will focus on three specific points of feedback:

1.	The Proposal should be voted on by token holders, excluding Kik and the Kin Foundation;
2.	The Quality Score should be refined to better empower developers to deliver a great user experience;
3.	The Proposal should eliminate the AUB cap.

**1.	Voting on the Proposal**

In the context of the Kin Rewards Engine, we believe the Kin Foundation should first and foremost serve as the facilitator of the iteration process, while also being empowered to make its own proposals for community consideration. Accordingly, we believe that any iterations to the KRE should only be implemented upon the vote of all interested token holders. We believe this vote should exclude Kik and the Kin Foundation as these entities were granted Kin tokens in the original token distribution, unlike all other token holders who had to buy or earn their Kin tokens.

**2.	Refine the Quality Score**

From our perspective, an effective Kin Rewards Engine is one that drives demand for Kin while maintaining fairness and simplicity for developers. In general, our concern is that inclusion of a Quality Score metric could introduce complexity and our preference would be that the KRE not include a Quality Score. That said, if the KRE utilizes the concept of a Quality Score, we believe three of the proposed required elements could undermine a developer’s ability to deliver a strong user experience and should be eliminated: (a) the buy module, (b) a standardized Kin tutorial, and (c) connecting an external wallet.

First, the integration of a buy-module could help drive real demand and provide users with an option to purchase Kin in-app. The challenge is that the options currently available are unreliable. Many developers have tested these buy modules and have had low success rates, which leads to a poor user experience. The success rate of these buy modules is directly correlated to the user’s location, which can yield varying results. We believe the intention is good (to drive more buy demand) but that this is already captured in the core AUB metric, and prescribing the method for purchase could be a net negative if developers have to implement a sub-par product. 

Second, the introduction of a standardized Kin tutorial could create confusion for users. In our view, the use of Kin should be left to the developer to address in their app how they see fit. Each app has a unique purpose and design and implementing a standardized tutorial could hurt the user experience.

Third, the requirement to add the ability to import or connect an external wallet may not align with the product flow of an app. Products may be predicated on utilizing a single wallet and shouldn't be penalized for that decision. It may also create additional security risks and access points for the individual user holding a large balance of Kin.

For these reasons, although we would prefer that there not be a Quality Score metric associated with the KRE, if one is adopted, we would recommend removing the questions noted above from the Quality Score questionnaire.

**3.	Eliminate AUB Cap**

​​​​Lastly, while we feel that the latest Proposal considers important new measures such as median spends and median balances, we feel that the $10 AUB cap should be removed.

Originally, implementation of the AUB cap was intended to deter developers from putting their KRE payments into their app in order to artificially inflate their average AUB. We witnessed this firsthand in prior iterations of the KRE. The 4.0 Proposal helps to solve this issue by moving away from average AUB and using median AUB instead. We feel this will largely eliminate the impact of outliers, and therefore obviates the need for a cap. This is preferable as we want apps driving as high an AUB as possible in each of their apps. Given this, we recommend removing the cap.

**Conclusion**

As a developer in the Kin Ecosystem, we at Code support the Kin Foundation’s commitment to iterating and improving the KRE. Successive iterations of the KRE that are supported by the entire Kin Community and bring more simplicity and fairness will contribute to achieving the mission of building a more fair digital world, with developers, consumers, and content-creators at the center.

**Comment by skyf0xx** [comment]
Eliminating AUB cap makes sense. The median balances would easily keep it under control.

There is also another potential way to simplify it while maintaining the core goals:
1. High balances for most users
2. High spends
3. Plenty of spends

This proposal on [Net Economic Activity](https://medium.com/kinblog/examination-of-recent-kre-improvement-proposals-edb9cb5cbcf7)  had been shelved *because* it would cause apps to maximize spend sizes. If this is the goal, NEA would solve for all the current desires while keeping the equation much simpler and easier to understand.

NEA could further be improved by using the median balance instead of AUB as originally proposed. Also, using the median sizes of economic spends for NEA would secure it even furthur.


In a nutshell, it's a much simpler formula, it encourages for high spends and needs fewer constraints to work (like enforcing a minimum spend amount).

**Reply by skyf0xx** [reply]
I have refined the idea over here
 https://github.com/kinecosystem/rewards-engine/discussions/170

**Comment by skyf0xx** [comment]
Added some suggestion on the quality score here https://github.com/kin-labs/rewards-engine/discussions/171
